misc/peer-store/src/memory_store.rs

```diff
@@ -81,6 +77,16 @@ impl<T> MemoryStore<T> {
     /// Remove an address record.
     /// Returns `true` when the address is removed.
     pub fn remove_address(&mut self, peer: &PeerId, address: &Multiaddr) -> bool {
+        let is_updated = self.remove_address_silent(peer, address);
+        if is_updated {
+            self.push_event_and_wake(crate::store::Event::RecordUpdated(*peer));
+        }
+        is_updated
+    }
+
+    /// Remove an address record without notifying swarm.
+    /// Returns `true` when the address is removed.
+    pub fn remove_address_silent(&mut self, peer: &PeerId, address: &Multiaddr) -> bool {

```

Comment:
```suggestion
    fn remove_address_silent(&mut self, peer: &PeerId, address: &Multiaddr) -> bool {

```
For the sake of simplicity I think only one method for removing addresses should be exposed to users.